### PR TITLE
docs: add a docstring to the check_sig method

### PR DESCRIPTION
**Pull request by CjS77**
Description
---

Helps clarify the algorithm and point out some edge cases that are being handled. Also the return type is not 100% intuitive, so this makes this clear too.

### CODE REVIEW

**Comment by CjS77** [comment]
ACK

File: infrastructure/tari_script/src/script.rs [hunk 1]
```diff
@@ -505,6 +505,22 @@ impl TariScript {
...
+    /// signatures. If the validation was NOT successful, `check_multisig` returns `Ok(None)`. This is a private
+    /// function, and callers will interpret these results according to their use cases.
+    ///
+    /// Other problems, such as stack underflows, invalid parameters etc return an `Err` as usual.
+    ///
+    /// Notes:
+    /// * The _m_ signatures are expected to be the top _m_ items on the stack.
+    /// * Every public key can be used AT MOST once.
+    /// * Every signature MUST be a valid signature using one of the public keys
+    /// * _n_ and _m_ must be positive AND n <= m AND n <= MAX_MULTISIG_LIMIT (32).
```
Comment (delta1) [inline infrastructure/tari_script/src/script.rs]:
```suggestion
    /// * _m_ and _n_ must be positive AND m <= n AND n <= MAX_MULTISIG_LIMIT (32).
```

Comment (delta1) [reply infrastructure/tari_script/src/script.rs]: from what I can see (and recall) in the code anyway :P

**Review by delta1 — APPROVED** [review]
ACK 2693b01

File: infrastructure/tari_script/src/script.rs (hunk 1 above)
Comment (CjS77) [reply infrastructure/tari_script/src/script.rs]: derp

**Comment by delta1** [comment]
PS: title slug should be `docs:`

**Review by sdbondi — COMMENTED** [review]
ACK

**Pull request merged**